src/main/webapp/pages/bookmarkedPlaceGuides_initPage.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * This function initialises the map and adds some functionalities to it:
+ * Geolocation: to track the user's current location, display it and center
+ * the map around it.
+ * Searching: lets the user search for places and center the map around it.
+ * Display place guides: displays the place guides created
+ * by the currently logged in user.
+ */
+let map;
```

Comment:
Yes, that may indeed cause unexpected bugs. However, I decided to postpone the removal of these global variables because I am planning some major changes in the organisation of the pages' content, which will probably affect the scope of these variables too and will make the removal easier.

Reply:
Ok SG.